components/formats-bsd/src/loci/formats/in/OBFReader.java

```diff
@@ -168,56 +191,76 @@ protected void initFile(String id) throws FormatException, IOException
+        final String name = obf.seriesMetadata.get("Name").toString();
+        ome.setImageName(name, series);
 
-      @SuppressWarnings("unchecked")
-      final List<Double> lengths = (List<Double>) obf.seriesMetadata.get("Lengths");
+        @SuppressWarnings("unchecked")
+        final List<Double> lengths = (List<Double>) obf.seriesMetadata.get("Lengths");
 
-      if (lengths.size() > 0)
-      {
-        double lengthX = Math.abs(lengths.get(0)) ;
-        if (lengthX < 0.01)
+        if (lengths.size() > 0)
         {
-          lengthX *= 1000000;
-        }
-        if (lengthX > 0)
-        {
-          final Length physicalSizeX = FormatTools.createLength( lengthX / obf.sizeX , UNITS.MICROM);
-          ome.setPixelsPhysicalSizeX(physicalSizeX, series);
+          double lengthX = Math.abs(lengths.get(0)) ;
+          if (lengthX < 0.01)
+          {
+            lengthX *= 1000000;
+          }
+          if (lengthX > 0)
+          {
+            final Length physicalSizeX = FormatTools.createLength( lengthX / obf.sizeX , UNITS.MICROM);
+            ome.setPixelsPhysicalSizeX(physicalSizeX, series);
+          }
         }
-      }
-      if (lengths.size() > 1)
-      {
-        double lengthY = Math.abs(lengths.get(1));
-        if (lengthY < 0.01)
+        if (lengths.size() > 1)
         {
-          lengthY *= 1000000;
+          double lengthY = Math.abs(lengths.get(1));
+          if (lengthY < 0.01)
+          {
+            lengthY *= 1000000;
+          }
+          if (lengthY > 0)
+          {
+            final Length physicalSizeY = FormatTools.createLength( lengthY / obf.sizeY , UNITS.MICROM);
+            ome.setPixelsPhysicalSizeY(physicalSizeY, series);
+          }
         }
-        if (lengthY > 0)
+        if (lengths.size() > 2)
         {
-          final Length physicalSizeY = FormatTools.createLength( lengthY / obf.sizeY , UNITS.MICROM);
-          ome.setPixelsPhysicalSizeY(physicalSizeY, series);
+          double lengthZ = Math.abs(lengths.get(2));
+          if (lengthZ < 0.01)
+          {
+            lengthZ *= 1000000 ;
+          }
+          if (lengthZ > 0)
+          {
+            final Length physicalSizeZ = FormatTools.createLength( lengthZ / obf.sizeZ  , UNITS.MICROM);
+            ome.setPixelsPhysicalSizeZ(physicalSizeZ, series);
+          }
         }
       }
-      if (lengths.size() > 2)
+    }
+    else
+    {
+      try 
       {
-        double lengthZ = Math.abs(lengths.get(2));
-        if (lengthZ < 0.01)
-        {
-          lengthZ *= 1000000 ;
-        }
-        if (lengthZ > 0)
-        {
-          final Length physicalSizeZ = FormatTools.createLength( lengthZ / obf.sizeZ  , UNITS.MICROM);
-          ome.setPixelsPhysicalSizeZ(physicalSizeZ, series);
-        }
+        ServiceFactory factory = new ServiceFactory() ;
+        OMEXMLService service = factory.getInstance( OMEXMLService.class ) ;
+
+        service.convertMetadata( ome_xml, ome ) ;
```

Comment:
See https://ci.openmicroscopy.org/job/BIOFORMATS-5.1-merge-test_images_good/648/console. This line generated a `NullPointerException` for some `msr` data. Excluding for a rebuild.